lib/record_store/provider/dnsimple/patch_api_header.rb

```diff
@@ -19,12 +19,17 @@ def execute(method, path, data = nil, options = {})
 
     private
 
-    def rate_limit_sleep(rate_limit_reset, rate_limit_remaining)
+    def rate_limit_sleep(_rate_limit_reset, _rate_limit_remaining)
       rate_limit_reset_in = [0, rate_limit_reset - Time.now.to_i].max
       rate_limit_periods = rate_limit_remaining + 1
-      wait_time = rate_limit_reset_in / rate_limit_periods.to_f
+      sleep_time = rate_limit_reset_in / rate_limit_periods.to_f
 
-      sleep(wait_time) if wait_time > 0
+      while sleep_time > 0
```

Comment:
Looks like this may be duplicated with the same loop in the NS1 patch.

Is it possible to extract this into a Module, or a Class?

Reply:
Should this be put into some separate `utils` folder in `record_store/provider/`?

Reply:
Could be =) Looks like functionality that would want to be shared across all providers, so that makes sense.

Consider also dumping it directly into `record_store/provider` if you don't want to create the folder/submodule. Just presenting alternatives.

Either way I trust your judgment on this one.